src/react/tests/useQuery.test.tsx

```diff
@@ -2552,6 +2553,114 @@ describe('useQuery', () => {
     consoleMock.mockRestore()
   })
 
+  it('should throw error if queryFn throws and useErrorBoundary is in use', async () => {
```

Comment:
Can you maybe add a test that checks if useErrorBoundary in the various settings is also respected now when suspense is true?

Reply:
Ok, I've added suspense.test.tsx tests for 3-4 different useErrorBoundary behaviours with suspense.